src/pages/basic-concepts/content-markdown.md

```diff
@@ -3,7 +3,143 @@ layout: ~/layouts/Main.astro
+
+`src/pages/index.md`
+
+```jsx
+---
+layout: ../layouts/main.astro
+---
+
+# Hello World!
+```
+
+Layout files are normal `.astro` components. Any Frontmatter defined in your `.md` page will be exposed to the Layout component as the `content` prop. `content` also has an `astro` key which holds special metadata about your file, like the complete Markdown `source` and a `headings` object.
+
+The rendered Markdown content is placed into the default `<Slot />` element.
+
+`src/layouts/main.astro`
+
+```jsx
+---
+export let content;
+---
+
+<html>
+  <head>
+    <title>{content.title}</title>
+  </head>
+
+  <body>
+    <slot/>
+  </body>
+</html>
+```
+
+### Markdown Component
+
+Similar to tools like [MDX](https://mdxjs.com/) or [MDsveX](https://mdsvex.com/), Astro makes it straightforward to embed rich, interactive components inside your Markdown content. The `<Markdown>` component is statically rendered, so it does not add any runtime overhead.
+
+Astro exposed a special `Markdown` component for `.astro` files which enables markdown syntax for its children **recursively**. Within the `Markdown` component you may also use plain HTML or any other type of component that is supported by Astro.
+
+```jsx
+---
+// For now, this import _must_ be named "Markdown" and _must not_ be wrapped with a custom component
+// We're working on easing these restrictions!
+import { Markdown } from 'astro/components';
+import Layout from '../layouts/main.astro';
+import MyFancyCodePreview from '../components/MyFancyCodePreview.tsx';
+
+const expressions = 'Lorem ipsum';
+---
+
+<Layout>
+  <Markdown>
+    # Hello world!
+
+    **Everything** supported in a `.md` file is also supported here!
+
+    There is _zero_ runtime overhead.
+
+    In addition, Astro supports:
+    - Astro {expressions}
+    - Automatic indentation normalization
+    - Automatic escaping of expressions inside code blocks
+
+    ```jsx
+      // This content is not transformed!
+      const object = { someOtherValue };
+    ```
+
+    - Rich component support like any `.astro` file!
+    - Recursive Markdown support (Component children are also processed as Markdown)
+
+    <MyFancyCodePreview:visible>
+      ```jsx
+      const object = { someOtherValue };
+      ```
+    </MyFancyCodePreview:visible>
+  </Markdown>
+</Layout>
+```
+
+### Remote Markdown
+
+If you have Markdown in a remote source, you may pass it directly to the Markdown component through the `content` attribute. For example, the example below fetches the README from Snowpack's Github repository and renders it as HTML.
+
+```jsx
+---
+import { Markdown } from 'astro/components';
+
+const content = await fetch('https://raw.githubusercontent.com/snowpackjs/snowpack/main/README.md').then(res => res.text());
+---
+
+<Layout>
+  <Markdown content={content} />
+</Layout>
+```
+
+Some times you might want to combine dynamic markdown with static markdown. You can nest `Markdown` components to get the best of both worlds.
```

Comment:
This could be:

There might be times where you would want to combine both, dynamic markdown with static markdown. In situations like this you can **nest** `<Markdown>` components with each other to get the best out of both worlds.